openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/KpiRepository.java

```diff
@@ -169,6 +171,14 @@ private EntityReference getDataInsightChart(Kpi kpi) throws IOException {
     return getToEntityRef(kpi.getId(), Relationship.USES, DATA_INSIGHT_CHART, true);
   }
 
+  public void validateDataInsightChartMapping(UUID chartId) {
+    // Each Chart has one unique Kpi mapping
+    List<CollectionDAO.EntityRelationshipRecord> record = findFrom(chartId, DATA_INSIGHT_CHART, Relationship.USES, KPI);
+    if (record.size() > 0) {
```

Comment:
if particular api has any chart associated that means size > 0 , then more kpi cannot be created for that charts, but if there are no chart to kpi mapping that is in case record == 0 , we should be able to create kpis, updating the function name